rfc/text/advanced/ap_27_pubsub_event_history.md

```diff
@@ -1,58 +1,158 @@
+    }
+  ]
+```
+
+In cases when the events set is pretty big to send it in a single result, router implementations
+may provide additional options, like pagination or returning a progressive results. 
+
+As Event History feature operates on `subscription|id` there can be situations when there is no subscribers to topic
+of interest yet, but publications happens. In this case the *Broker* can not prematurely know what events to store.
+If the *Broker* implementation allows configuration on per-topic basis, it may overcome this situations by 
+preinitializing history-enabled topics with subscriptions even if there is no real subscribers yet exists.
+
+Sometimes a client may not be interested in subscribing to a topic just to get a subscription id. In that case
+a client may use some [Subscriptions Meta API RPC](#name-procedures-3) for retrieving subscription IDs by topic URIs
+if WAMP router supports it.
+
+**Security Aspects**
+
+TODO/FIXME: This part of Event History needs more discussion and clarification.
+But at least provides some basic information to take into account.
+
+To be able to request event history, peer must be allowed to subscribe to desired subscription first. Thus, if peer
+can not subscribe to a topic (which results in subscription under the hood) it can not receive events history too. 
+And second point: peer must be allowed to call related META procedures for getting the event history described above.
+Event History RPC calls with prohibited requests must fail with `wamp.error.not_authorized` Error URI.
+
+Original publications may include additional options, like `black-white-listing` that enforces special event 
+processing. The same rules must apply to event history requests. For example, if original publication has 
+`eligible_authrole = 'admin'`, but request for history came from peer with `authrole = 'user'`, then even if 
+`user` is authorized to subscribe to topic and thus is authorized to ask for event history, this publication 
+must be filtered out from this concrete request results on the router side.
+
+`black-white-listing` feature also allows to filter events delivery on `session ID` basis. In the context of
+event history that can results in unexpected behaviour: `session ID` are generated randomly in runtime for every
+session connection so newly connected sessions asking for event history may receive events originally excluded 
+or vice versa may not receive expected events due to session ID mismatch. To prevent this unexpected behaviour
+all events published with `Options.exclude|list[int]` or `Options.eligible|list[int]` should be ignored by event
+history implementation, mean not saved at all.
+
+To wrap it up: event history may operate on rather stable session attributes, for now it is `authrole` and `authid`,
+all dynamic attributes like `session ID` or maybe other custom attributes in future should lead to ignore storing
```

Comment:
```suggestion
Filtering based on ephemeral attributes like `session ID` – and perhaps other future custom attributes – should result in the event not being stored in the history at all, to avoid unintentional leaking of event information.
```